authors/apps/comments/views.py

```diff
@@ -0,0 +1,142 @@
+        if not comment:
+            return Response({
+                "error": "This article does not have this comment"
+            }, status.HTTP_404_NOT_FOUND)
+        current_user = request.user
+        comment = Comment.objects.filter(article=article, id=comment_id, author=current_user).first()
+        if not comment:
+            return Response({
+                "error": "You can't edit someone else's comment"
+            }, status.HTTP_401_UNAUTHORIZED)
+        data = request.data.get('comment', {})
+        context = {"author": request.user, "article": article}
+        serializer = self.serializer_class(comment, data=data, context=context, partial=True)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        response = {
+            "message": "Your comment has been updated",
+            "comment": serializer.data
+        }
+        return Response(response, status.HTTP_201_CREATED)
+
+    def delete(self, request, *args, **kwargs):
+        slug = kwargs.get('slug')
+        comment_id = kwargs.get('pk')
+        article = self.get_article(slug)
+        comment = Comment.objects.filter(article=article, id=comment_id).first()
+        if not comment:
+            return Response({
+                "error": "This article does not have this comment"
+            }, status.HTTP_404_NOT_FOUND)
+        current_user = request.user
+        comment = Comment.objects.filter(article=article, id=comment_id, author=current_user).first()
+        if not comment:
+            return Response({
+                "error": "You can't delete someone else's comment"
+            }, status.HTTP_401_UNAUTHORIZED)
+        serializer = self.serializer_class(comment, partial=True)
+        comment.delete()
+        response = {
+            "message": "Your comment has been deleted",
+            "comment": serializer.data
+        }
+        return Response(response, status.HTTP_201_CREATED)
+   
+    def get(self, request, **kwargs):
+        slug = kwargs.get('slug')
+        comment_id = kwargs.get('pk')
+        article = self.get_article(slug)
+        queryset = Comment.objects.filter(article=article, id=comment_id)
+        if not queryset:
+            return Response({
+                "error": "This article does not have this comment"
+            }, status.HTTP_404_NOT_FOUND)
+        serializer = self.serializer_class(queryset, many=True)
+        return Response(serializer.data, status.HTTP_200_OK)
+
+    def get_article(self, slug):
+        try:
+            article = get_object_or_404(Article, slug=slug)
+        except:
+            raise exceptions.ValidationError("This article does not exist!")
+        return article
+
+    def get_comment(self, comment_id):
+        try:
+            comment = get_object_or_404(Comment, id=comment_id)
+        except:
+            raise exceptions.ValidationError("This article does not have this comment!")
```

Comment:
Please update this error message accordingly as well, as shown before

Reply:
I have resolved this.